lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -2948,11 +3006,128 @@ ::mlir::LogicalResult mlir::tt::ttir::GenericOp::verify() {
             "Additional GenericOp region arguments must be of SemaphoreType");
       }
     }
+
+    auto const *firstIndexingMap = getIndexingMaps().begin();
+    for (auto &indexingMap : getIndexingMaps()) {
+      if (mlir::cast<AffineMapAttr>(indexingMap).getValue().getNumDims() !=
+          mlir::cast<AffineMapAttr>(*firstIndexingMap)
+              .getValue()
+              .getNumDims()) {
+        return emitOpError("all indexing maps must have the same number of "
+                           "dimensions");
+      }
+    }
```

Comment:
Nit (kind of more readable, both are not ideal):
```suggestion
    if (!llvm::all_equal(llvm::map_range(getIndexingMapsValue(), [](AffineMap m) {
          return m.getNumDims();
        }))) {
      return emitOpError("all indexing maps must have the same number of "
                         "dimensions");
    }
```

Reply:
For some strange reason this doesn't work.. looking into it 🤔